libraries/ui-library/src/components/six-header/six-header.tsx

```diff
@@ -194,27 +196,42 @@ export class SixHeader {
     };
 
     if (this.hasSlot(Slot.AppSwitcher)) {
-      this.selectedApp = this.getSelectedApp();
+      this.updateSelectedApp();
     }
 
     if (this.hasSlot(Slot.Search)) {
       this.computeSearchOpenState();
     }
   }
 
-  private getSelectedApp(): string | undefined {
+  componentDidLoad() {
+    const appSwitcherSlot = getSlot(this.host, Slot.AppSwitcher);
```

Comment:
It's probably better to initialize the mutation observer in the `connectedCallback` rather than in `componentDidLoad`. This is because once a component is disconnected, `componentDidLoad` won't be called again if the component reconnects. For more details, refer to the section on `connectedCallback` in the StencilJS Component Lifecycle documentation: https://stenciljs.com/docs/component-lifecycle#connectedcallback

Reply:
makes sense, i changed this 👍 
i thought about changing it in `six-tab-group` too, but there is a lot more initialization going on in `componentDidLoad` and i did not want to risk introducing new bugs through this fix ;)